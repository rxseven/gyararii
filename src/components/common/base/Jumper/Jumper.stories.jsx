import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import Jumper from './index';

const { text } = mock;
const title = 'Components/Common/Base/Jumper';

storiesOf(title, module).add('default', () => (
  <div>
    <p>{text}</p>
    <p>{text}</p>
    <p>{text}</p>
    <p>{text}</p>
    <p>{text}</p>
    <p>{text}</p>
    <p>{text}</p>
    <p>{text}</p>
    <Jumper />
  </div>
));
