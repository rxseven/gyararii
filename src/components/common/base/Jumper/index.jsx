import React from 'react';
import ScrollToTop from 'react-scroll-up';
import styled from 'styled-components';

import Icon from 'components/common/base/Icon';

const Arrow = styled(Icon)`
  color: rgba(0, 0, 0, 0.35);
  font-size: 1.35rem;
  transition: color 0.2s ease-in-out;

  :hover {
    color: rgba(0, 0, 0, 0.65);
  }

  @media (min-width: ${({ theme }) => theme.breakpoint.sm}) {
    color: rgba(0, 0, 0, 0.6);
    font-size: 1.15rem;

    :hover {
      color: rgba(0, 0, 0, 0.75);
    }
  }
`;

const Frame = styled.div`
  align-items: center;
  background-color: rgba(255, 255, 255, 0.5);
  border: rgba(0, 0, 0, 0.25) solid 1px;
  border-radius: 50%;
  bottom: 1rem;
  display: flex;
  height: 35px;
  justify-content: center;
  margin-right: ${() => parseFloat(-35 / 2)}px;
  position: fixed;
  right: 50%;
  transition: border-color 0.2s ease-in-out;
  width: 35px;
  z-index: 1;

  :hover {
    border-color: rgba(0, 0, 0, 0.4);
  }

  @media (min-width: ${({ theme }) => theme.breakpoint.sm}) {
    background-color: rgba(255, 255, 255, 1);
    border-color: rgba(0, 0, 0, 0.35);
    height: 30px;
    margin-right: 0;
    right: 1.5rem;
    width: 30px;

    :hover {
      border-color: rgba(0, 0, 0, 0.5);
    }
  }
`;

const options = {
  showUnder: 160,
  style: {
    bottom: 0,
    left: 0,
    position: 'absolute',
    right: 0
  }
};

function Jumper() {
  return (
    <ScrollToTop {...options}>
      <Frame>
        <Arrow icon="arrow-up" />
      </Frame>
    </ScrollToTop>
  );
}

export default Jumper;
