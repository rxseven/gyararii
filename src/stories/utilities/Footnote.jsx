import styled from 'styled-components';

const Footnote = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

export default Footnote;
