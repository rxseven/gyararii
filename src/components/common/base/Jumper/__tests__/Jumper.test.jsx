import { factory } from 'tests/utilities';
import Jumper from '../index';

// Arrange
const source = {};

// Setup
function setup(props) {
  return factory(Jumper, source, props);
}

// Test suites
describe('<Jumper />', () => {
  it('should render without crashing', () => {
    setup();
  });
});
