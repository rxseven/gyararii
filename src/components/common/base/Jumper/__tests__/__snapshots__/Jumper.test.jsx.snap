// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumper /> Snapshot tests should render correctly 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "bottom": 0,
      "cursor": "pointer",
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "right": 0,
      "transitionDelay": "0s",
      "transitionDuration": "0.2s",
      "transitionProperty": "opacity, visibility",
      "transitionTimingFunction": "linear",
      "visibility": "hidden",
    }
  }
>
  <div
    className="sc-bwzfXH dobsVM"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-arrow-up fa-w-14 sc-bdVaJa bRiyoa"
      data-icon="arrow-up"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
